operator/src/main/java/org/bf2/operator/managers/IngressControllerManager.java

```diff
@@ -432,6 +449,66 @@ void reconcileIngressControllers() {
         }
     }
 
+    public void ensureBlueprintRouteMatching(Route route, String blueprintBaseName) {
+        var annotations = Optional.ofNullable(route.getMetadata()).map(ObjectMeta::getAnnotations);
+        var tslConfig = Optional.ofNullable(route.getSpec()).map(RouteSpec::getTls);
+
+        ensureBlueprintRouteMatching(annotations, tslConfig, blueprintBaseName);
+    }
+
+    public void ensureBlueprintRouteMatching(Optional<Map<String, String>> annotations, Optional<TLSConfig> tlsConfig, String blueprintBaseName) {
+        // see findMatchingBlueprint https://github.com/openshift/router/blob/master/pkg/router/template/configmanager/haproxy/manager.go#L817
+        if (!Boolean.TRUE.equals(dynamicConfigManager) || blueprintRouteNamespace == null) {
+            return;
+        }
+
+        var orderedAnnotations = annotations.map(TreeMap::new);
+        MessageDigest instance;
+        try {
+            instance = MessageDigest.getInstance("SHA-1");
+            orderedAnnotations.ifPresent(m -> m.forEach((k, v) -> {
+                instance.update(String.valueOf(k).getBytes());
+                instance.update(String.valueOf(v).getBytes());
+            }));
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
+        }
+        tlsConfig.map(TLSConfig::getTermination).map(String::getBytes).ifPresent(instance::update);
+        tlsConfig.map(TLSConfig::getCertificate).map(String::getBytes).ifPresent(instance::update);
+        tlsConfig.map(TLSConfig::getKey).map(String::getBytes).ifPresent(instance::update);
+        tlsConfig.map(TLSConfig::getCaCertificate).map(String::getBytes).ifPresent(instance::update);
```

Comment:
Is the order that things are added to the digest important?

Reply:
No, providing the order is stable the digest will be the same.  It relies on a stable name otherwise we'll see a accumulation of routes.